cmd/sys-mgmt-executor/main.go

```diff
@@ -58,109 +53,36 @@ func main() {
-			Running bool
-		}
-	}
-
-	c := []containerInfo{}
-	for {
-		err = dec.Decode(&c)
-		if err != nil {
-			return false, err
-		}
-		switch {
-		case len(c) < 1:
-			return false, fmt.Errorf("container %s not found", service)
-		case len(c) > 1:
-			return false, fmt.Errorf("multiple containers found with name %s", service)
-		default:
-			if c[0].State.Running {
-				LoggingClient.Info(fmt.Sprintf("service container %s is running", service))
+		service := os.Args[1]
+		operation := os.Args[2]
+
+		// Don't run commands for unknown services - could be an attack
+		if agent.IsKnownServiceKey(service) {
+			result, err := executor.Execute(operation, service, func(arg ...string) ([]byte, error) {
+				return exec.Command("docker", arg...).CombinedOutput()
+			})
+			if err != nil {
+				LoggingClient.Error(fmt.Sprintf("error performing %s on service %s: %v", operation, service, err.Error()))
 			} else {
-				LoggingClient.Info(fmt.Sprintf("service container %s is not running", service))
+				LoggingClient.Info(fmt.Sprintf("success performing %s on service %s", operation, service))
+				LoggingClient.Debug(fmt.Sprintf("the IsKnownServiceKey() check was a success for service %s", service))
+				LoggingClient.Debug(fmt.Sprintf("operation: %s", operation))
+
+				if result != nil {
+					// Push result to stdout for consumption by system management agent
+					fmt.Printf("%s", string(result))
+				}
 			}
-			return c[0].State.Running == running, err
-		}
-	}
-}
-
-func ExecuteDockerCommands(service string, operation string) error {
-	// don't run commands for unknown services - could be an attack
-	if agent.IsKnownServiceKey(service) {
-		return runDockerCommands(service, operation)
-	}
-	return fmt.Errorf("the service %s is an unknown service for which request was made to run Docker command", service)
-}
-
-func runDockerCommands(service string, operation string) error {
-	// Validate that a known operation was requested.
-	if operation == START || operation == STOP || operation == RESTART {
-		// run the docker command
-		out, err := exec.Command("docker", operation, service).CombinedOutput()
-		if err != nil {
-			LoggingClient.Error(fmt.Sprintf("docker command failed: %s", string(out)))
-			return err
-		}
-
-		// check that the command actually resulted in the correct state for
-		// the container
-		var expectedStatus bool
-		switch operation {
-		case START:
-			fallthrough
-		case RESTART:
-			expectedStatus = true
-		case STOP:
-			expectedStatus = false
-		default:
-			// impossible to get here
-			panic(fmt.Sprintf("invalid operation %s", operation))
-		}
-
-		correctStatus, err := checkDockerContainerStatus(service, expectedStatus)
-		if err != nil {
-			return err
-		}
-		if !correctStatus {
-			return fmt.Errorf("docker %s operation failed for service %s", operation, service)
+		} else {
+			LoggingClient.Error(fmt.Sprintf("the service %s is an unknown one for which the request was made to run Docker command", service))
 		}
```

Comment:
It occurs to me that this app should utilize os.Exit() as the other services do. If exiting based on successful completion, use `os.Exit(0)`. If terminating due to error, use `os.Exit(1)`